core/src/main/java/com/rudderstack/android/sdk/core/RudderMessageBuilder.java

```diff
@@ -74,13 +75,23 @@ public RudderMessageBuilder setUserProperty(Map<String, Object> map) {
     }
 
     // TODO:  need to figure out to use it as integrations
-    private RudderOption option;
+    private Map<String, Object> option;
 
-    public RudderMessageBuilder setRudderOption(RudderOption option) {
-        this.option = option;
+    public RudderMessageBuilder setRudderOption(Map<String, Object> option) {
+        this.option = new LinkedHashMap<String,Object>();
```

Comment:
can we use `HashMap` instead of `LinkedHashMap` ?

Reply:
done